common/Util-desktop.cpp

```diff
@@ -160,6 +162,56 @@ std::size_t getTotalSystemMemoryKb()
     return totalMemKb;
 }
 
+size_t getMaxConcurrentTCPConnections(const size_t minimum)
+{
+#ifdef __linux__
+    auto readDecimal = [](const char* path, const size_t defaultValue) -> size_t {
+        const int fd = ::open(path, O_RDONLY);
+        if (fd<0)
+            return defaultValue;
+        char data[1024+1]; // includes EOS
+        ::memset(data, 0, sizeof(data));
+        size_t consumed = 0;
+        while (sizeof(data)-consumed > 1 /* ex-EOS */)
+        {
+            ssize_t n;
+            while ((n = ::read(fd, data+consumed, sizeof(data)-consumed-1 /* ex-EOS */)) < 0 && errno == EINTR) {}
+            if (n < 0) // Error
+            {
+                consumed = 0;
+                break;
+            }
+            else if (n == 0) // EOF
+                break;
+            consumed += size_t(n);
+        }
+        ::close(fd);
+        if (consumed==0)
+            return defaultValue;
+        char* endptr = nullptr;
+        errno = 0;  // Flush previous error indicator. Reminder: errno is thread-local
+        const unsigned long long num = std::strtoull(data, &endptr, 10);
+        if (0 != errno || nullptr == endptr || endptr == data || num > std::numeric_limits<size_t>::max())
+            return defaultValue;
+        return size_t(num);
+    };
+    // - 4 tcp_max_orphans sockets per MB
+    // - {4096M -> 16384}, {16384M -> 65536}, {65407M -> 262144}, ...
+    const ssize_t tcp_max_orphans = readDecimal("/proc/sys/net/ipv4/tcp_max_orphans", 0); // ignored if n/a
```

Comment:
[tcp_max_orphans](https://www.man7.org/linux/man-pages/man7/tcp.7.html) is that memory bound freely available socket count, not getting reset. 
> This limit exists only to prevent simple denial-of-service attacks.  Lowering this limit is not recommended. Network conditions might require you to increase the number of orphans allowed, but note that each orphan can eat up to ~64 kB of unswappable memory.  The default initial value is set equal to the kernel parameter NR_FILE.  This initial default is adjusted depending on the memory in the system.

While more sockets can be used, it is a good indicator of a worry free socket count IMHO _at_ application launch.
`nf_conntrack_max` is usually set to `tcp_max_orphans` by default, if the iptables conntrack module is loaded.

Here emphasis should be on _concurrent_tcp_sockets_ and whether we like to use this metric to limit external connections in the first place, I understand this.

It may make sense to also inject an alternative `max_tcp_connections` COOLWSD configuration value.
However cropping such value to `nf_conntrack_max` makes little sense, i.e. it implies the admin has to adjust the `nf_conntrack_max` _and_ (probably) `tcp_max_orphans` limits anyways.
Hence we could use a config COOLWSD  `max_tcp_connections`, use it as-is (if given) but warn if the system `getMaxConcurrentTCPConnections` is lower (-> for the system admin)?

Now if we look at the memory based Linux defaults here, i.e. 4 sockets/MB or 262k sockets for 64G etc, 
it looks to me that these are good values. Whenever this issue is discussed in the web server space, 
it is said to increase the same system limits to achieve a higher concurrent responsiveness while 
(need to) maintain the DoS balance - if that makes sense.

Reply:
The setting is some random other number - for a different use-case. It may often be initialized to something sensible - but it's really for a different purpose AFAICS.

So - fine; lets do this guess-timate thing - but we need to provide a setting to override it, in the case that someone really wants to configure their orphans in some way - and not crop the number of clients connections we can support.

Reply:
Added WSD config value evaluation.